OSDOCS#5338: Scale down data plane to zero (#63784)
openshift/openshift-docs: modules/scale-down-data-plane.adoc

// Module included in the following assemblies:
//
// * hosted_control_planes/hcp-managing.adoc

:_content-type: PROCEDURE
[id="scale-down-data-plane_{context}"]
= Scaling down the data plane to zero

If you are not using the hosted control plane, to save the resources and cost you can scale down a data plane to zero.

.Procedure

. Set the `kubeconfig` file to access the hosted cluster by running the following command:
+
[source,terminal]
----
$ export KUBECONFIG=<install_directory>/auth/kubeconfig
----

> . Get the name of `NodePool` associated to your hosted cluster by running the following command:

Reviewer: ```suggestion
. Get the name of the `NodePool` associated to your hosted cluster by running the following command:
```

+
[source,terminal]
----
$ oc get nodepool --namespace <HOSTED_CLUSTER_NAMESPACE>
----

. Optional: To prevent the pods from draining, add the `nodeDrainTimeout` field in the `NodePool` resource by running the following command:
+
[source,terminal]
----

> $ oc edit NodePool <nodepool> -o yaml --namespace <HOSTED_CLUSTER_NAMESPACE>

Reviewer 2: fwiw you can also `oc scale NodePool <nodepool> --namespace <HOSTED_CLUSTER_NAMESPACE> --replicas 0`

Reviewer 2: ah nvm I see you have that below.

----
+
.Example output
[source,terminal]
----
spec:
  arch: amd64
  clusterName: clustername <1>
  management:
    autoRepair: false
    replace:
      rollingUpdate:
        maxSurge: 1
        maxUnavailable: 0
      strategy: RollingUpdate
    upgradeType: Replace
  nodeDrainTimeout: 0s <2>
----
<1> Defines the name of your hosted cluster.

> <2> Specifies the total amount of time that the controller spends to drain a node. The default value is `0` seconds. By default, the `nodeDrainTimeout: 0s` setting blocks the node draining process.

Reviewer: ```suggestion
<2> Specifies the total amount of time that the controller spends to drain a node. By default, the `nodeDrainTimeout: 0s` setting blocks the node draining process.
```
The last sentence specifies the default value is `0` seconds, so I dont think you need that additional sentence.

+
[NOTE]
====

> If you want to allow the node draining process for a certain period of time, you can set the value of the `nodeDrainTimeout` field accordingly, for example, `nodeDrainTimeout: 1m`.

Reviewer: Just want to highlight that "want" is a use with caution word in SSG https://redhat-documentation.github.io/supplementary-style-guide/#want 

```suggestion
You can set the value of the `nodeDrainTimeout` field accordingly, for example, `nodeDrainTimeout: 1m` to allow the node draining process to continue for a certain period of time.
```

====

> . Scale down `NodePool` associated to your hosted cluster by running the following command:

Reviewer: ```suggestion
. Scale down the `NodePool` associated to your hosted cluster by running the following command:
```

// Module included in the following assemblies:
//
// * hosted_control_planes/hcp-managing.adoc

:_content-type: PROCEDURE
[id="scale-down-data-plane_{context}"]
= Scaling down the data plane to zero

If you are not using the hosted control plane, to save the resources and cost you can scale down a data plane to zero.

[NOTE]
====
Ensure you are prepared to scale down the data plane to zero. Because the workload from the worker nodes disappears after scaling down.
====

.Procedure

. Set the `kubeconfig` file to access the hosted cluster by running the following command:
+
[source,terminal]
----
$ export KUBECONFIG=<install_directory>/auth/kubeconfig
----

. Get the name of the `NodePool` resource associated to your hosted cluster by running the following command:
+
[source,terminal]
----
$ oc get nodepool --namespace <HOSTED_CLUSTER_NAMESPACE>
----

. Optional: To prevent the pods from draining, add the `nodeDrainTimeout` field in the `NodePool` resource by running the following command:
+
[source,terminal]
----
$ oc edit NodePool <nodepool> -o yaml --namespace <HOSTED_CLUSTER_NAMESPACE>
----
+
.Example output

> [source,terminal]

Reviewer 3: Should this be [source, yaml]

additionally, I am used to seeing example resource output with a little more context at the beginning. LIke:
`apiVersion: v1alpha1
kind: NodePool
metadata:
  name: cluster`

Reviewer 3: Like Step 3 here:
https://docs.openshift.com/container-platform/4.13/scalability_and_performance/using-node-tuning-operator.html#node-tuning-hosted-cluster_node-tuning-operator

See how there's the API context at the top?

Author: Thanks! Addressed
